### PR TITLE
Fixing Types Errors

### PR DESCRIPTION
**Pull request by MuhammadHassan03**

### CODE REVIEW

**Review by andrew-bierman — CHANGES_REQUESTED** [review]
On the right path but let's fix the failing GitHub actions

File: packages/app/components/card/PackCardHeader/PackCardHeader.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ export const PackCardHeader = ({ data, title }: PackCardHeaderProps) => {
             </View>
           </RStack>
         }
+        link = {null}
```
[the remark this answers is not in the document]
Comment (MuhammadHassan03) [reply packages/app/components/card/PackCardHeader/PackCardHeader.tsx]: i am done with this i made it optional

File: packages/app/screens/user/Settings.tsx [hunk 2]
```diff
@@ -122,15 +122,15 @@ export default function Settings() {
                   <RLabel>Weather: </RLabel>
                   <FormSelect
                     options={weatherOptions}
-                    name="preferredWeather"
+                    name="value.preferredWeather"
```
Comment (pinocchio-life-like) [inline packages/app/screens/user/Settings.tsx:125]: Are we nesting object or some kind of deep linking of form field names to values? If not I believe the first should be okay.

**Comment by pinocchio-life-like** [comment]
Overall it looks good to me, but test the UI for failures thoroughly. Thank you!

**Comment by pinocchio-life-like** [comment]
@MuhammadHassan03 and this is why the cloudlfare is failing:
Screenshot 2024-06-27 111753

**Comment by MuhammadHassan03** [comment]
> @MuhammadHassan03 and this is why the cloudlfare is failing:
> Screenshot 2024-06-27 111753
> 
> 

Got it am working on it

**Comment by MuhammadHassan03** [comment]
@pinocchio-life-like i have made requested changes please check it out

File: apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx [hunk 3]
```diff
@@ -4,25 +4,22 @@ import { StyleSheet, Text, View, Platform } from 'react-native';
...
-  const userRealName = user?.name ?? null;
 
   return (
     <>
       {Platform.OS === 'web' && (
         <Head>
-          <title>{`${userRealName}'s Profile`}</title>
-          <meta name="description" content={`${userRealName}'s Profile`} />
+          <title>{`${id}'s Profile`}</title>
+          <meta name="description" content={`${id}'s Profile`} />
```
Comment (pinocchio-life-like) [inline apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx:18]: This is needs to be reverted, and the ones above and below as well.

**Comment by pinocchio-life-like** [comment]
@MuhammadHassan03 please address each comment and set as resolved if so or leave feedback.

**Review by pinocchio-life-like — CHANGES_REQUESTED** [review]

File: apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx (hunk 3 above)
Comment (MuhammadHassan03) [reply apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx:18]: i reverted it

File: packages/app/screens/user/Settings.tsx (hunk 2 above)
Comment (MuhammadHassan03) [reply packages/app/screens/user/Settings.tsx]: you are correct.. we aren't nesting it so i reverted it also

File: packages/app/hooks/chat/useChat.ts [hunk 4]
```diff
@@ -12,8 +16,8 @@ export const useChat = (itemTypeId = null) => {
   // const [parsedMessages, setParsedMessages] = useState([]);
 
   const { data: chatsData, refetch } = useGetUserChats(
-    user.id,
-    typeId.itemTypeId,
+    user?.id,
```
[the remark this answers is not in the document]
Comment (MuhammadHassan03) [reply packages/app/hooks/chat/useChat.ts]: got it and reverted

File: packages/app/components/chat/index.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ interface ChatComponentProps {
 interface ChatModalTriggerProps {
   title: string;
   trigger: string;
-  itemTypeId: string | null;
+  itemTypeId?: string | null;
```
[the remark this answers is not in the document]
Comment (MuhammadHassan03) [reply packages/app/components/chat/index.tsx]: reverted

File: packages/app/components/card/CustomCard.tsx [hunk 6]
```diff
@@ -73,7 +73,7 @@ export const CustomCard = ({
           )}
         </View>
         <RSeparator />
-        {type === 'pack' && authUser?.id === data.owner_id ? (
+        {type === 'pack' && authUser?.id === data.owner_id.id ? (
```
[the remark this answers is not in the document]
Comment (MuhammadHassan03) [reply packages/app/components/card/CustomCard.tsx:78]: yeah, am sure its a object according to the props interface in this file

File: apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx [hunk 7]
```diff
@@ -16,13 +15,13 @@ const Profile = () => {
       {Platform.OS === 'web' && (
         <Head>
           <title>{`${userRealName}'s Profile`}</title>
-          <meta name="description" content={`${userRealName}'s Profile`} />
+          <meta name="description" content={`${id}'s Profile`} />
         </Head>
       )}
       <Stack.Screen
+        name={`${id}'s Profile`}
```
Comment (pinocchio-life-like) [inline apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx:22]: this

File: apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx [hunk 8]
```diff
@@ -16,13 +15,13 @@ const Profile = () => {
...
         options={{
-          title: `${userRealName}'s Profile`,
-          name: `${userRealName}'s Profile`,
+          title: `${id}'s Profile`,
```
Comment (pinocchio-life-like) [inline apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx:24]: and this, to be consistent

Comment (MuhammadHassan03) [reply apps/expo/app/(app)/(drawer)/(tabs)/(stack)/profile/[id].tsx:24]: please check it out in this commit [1f17ce0]

**Comment by pinocchio-life-like** [comment]
The create trip page seems to be failing:

**Comment by pinocchio-life-like** [comment]
Ok got it.

**Review by pinocchio-life-like — APPROVED** [review]

**Comment by EngrAJ8888** [comment]
Hi @MuhammadHassan03 , I'm unable to proceed with creating a pack due to this error. Could you please take a look? Thanks! 
image

**Comment by MuhammadHassan03** [comment]
> Hi @MuhammadHassan03 , I'm unable to proceed with creating a pack due to this error. Could you please take a look? Thanks! image

@EngrAJ8888 it's fixed please check it out

**Comment by EngrAJ8888** [comment]
Hi @MuhammadHassan03, I encountered an error message while trying to create a pack. Please update, thanks! 
image

**Comment by MuhammadHassan03** [comment]
> Hi @MuhammadHassan03, I encountered an error message while trying to create a pack. Please update, thanks! image



> Hi @MuhammadHassan03, I encountered an error message while trying to create a pack. Please update, thanks! image



> Hi @MuhammadHassan03, I encountered an error message while trying to create a pack. Please update, thanks! image

it is working fine on my side please check it out in this recording
https://drive.google.com/file/d/1jqhV9x_CL2a69sRAneEq21ztLzVyGpOo/view?usp=sharing

**Comment by JewelTee** [comment]
> Hi @MuhammadHassan03, I encountered an error message while trying to create a pack. Please update, thanks! image

@MuhammadHassan03 I'm encountering this error on my end as well

**Comment by EngrAJ8888** [comment]
Hi @MuhammadHassan03 , we're using the preview URL https://feat-fix-types.packrat.pages.dev/, and it's not functioning correctly on the live version.

**Comment by MuhammadHassan03** [comment]
Got it i'll check it out

**Comment by andrew-bierman** [comment]
@MuhammadHassan03 merge testing into here, solved a few map file errors

**Comment by MuhammadHassan03** [comment]
Hey @EngrAJ8888 it's fixed now on preview url. Please Check it out

**Pull request merged**